<launch>
  <arg name="input/odometry" default="/localization/kinematic_state"/>
  <arg name="input/lanelet_map_bin" default="/map/vector_map"/>
  <arg name="input/route" default="/planning/mission_planning/route"/>
  <arg name="input/reference_trajectory" default="/planning/scenario_planning/trajectory"/>
  <arg name="input/predicted_trajectory" default="/control/trajectory_follower/predicted_trajectory"/>
  <arg name="config_file" default="$(find-pkg-share autoware_lane_departure_checker)/config/lane_departure_checker.param.yaml"/>

  <!-- vehicle info -->
  <arg name="vehicle_info_param_file" default="$(find-pkg-share autoware_vehicle_info_utils)/config/vehicle_info.param.yaml"/>

  <node pkg="autoware_lane_departure_checker" exec="lane_departure_checker_node" name="lane_departure_checker_node" output="screen">
    <remap from="~/input/odometry" to="$(var input/odometry)"/>
    <remap from="~/input/lanelet_map_bin" to="$(var input/lanelet_map_bin)"/>
    <remap from="~/input/route" to="$(var input/route)"/>
    <remap from="~/input/reference_trajectory" to="$(var input/reference_trajectory)"/>
    <remap from="~/input/predicted_trajectory" to="$(var input/predicted_trajectory)"/>
    <param from="$(var config_file)"/>
    <param from="$(var vehicle_info_param_file)"/>
  </node>
</launch>
